media/js/privacy/firefox-privacy.js

```diff
@@ -0,0 +1,288 @@
+            divElem.setAttribute('aria-hidden', true);
+
+            topicHeading.classList.add('collapsed');
+        }
+    }
+
+    /**
+     * Runs a set of functions on page load.
+     */
+    function initPage() {
+        collectAllTopics();
+        hideAllTopicContent();
+        addMasterToggles();
+        showInitialTopic();
+    }
+
+    /**
+     * On load, expands the first topic of the first main section. This
+     * also calls `addDataChoicesWidget` if on Fx desktop.
+     */
+    function showInitialTopic() {
+        var initialTopic = topicHeaders[0].querySelector('section');
+        var initialTopicContent = initialTopic.querySelector('div');
+        var initialTopicHeading = initialTopic.querySelector('h3');
+
+        initialTopicContent.classList.remove('hidden');
+        initialTopicContent.setAttribute('aria-hidden', false);
+
+        initialTopicHeading.classList.remove('collapsed');
+
+        if (Mozilla.Client.isFirefoxDesktop) {
+            addDataChoicesWidget(initialTopicContent);
+        }
+    }
+
+    /**
+     * Hide or show all topics for the main section.
+     * @param {Object} targetElem - The element that triggered the event
+     */
+    function toggleMainSectionTopics(targetElem) {
+        var targetSectionId = dataAttributeHandler({
+            elem: targetElem,
+            property: 'data-parent-container'
+        });
+        var isExpanded = dataAttributeHandler({
+            elem: targetElem,
+            property: 'data-is-expanded'
+        });
+
+        var targetAction = isExpanded === 'true' ? 'add' : 'remove';
+        var targetSection = document.getElementById(targetSectionId);
+        var subSections = targetSection.querySelectorAll('section');
+
+        for (var i = 0, l = subSections.length; i < l; i++) {
+            var divElem = subSections[i].querySelector('div');
+            var heading = subSections[i].querySelector('h3');
+
+            divElem.classList[targetAction]('hidden');
+            divElem.setAttribute('aria-hidden', isExpanded || 'false');
+            heading.classList[targetAction]('collapsed');
+        }
+
+        if (isExpanded === 'true') {
+            dataAttributeHandler({
+                elem: targetElem,
+                property: 'data-is-expanded',
+                value: 'false'
+            });
+
+            targetElem.textContent = tabpanelOpenText;
+        } else {
+            dataAttributeHandler({
+                elem: targetElem,
+                property: 'data-is-expanded',
+                value: 'true'
+            });
+
+            targetElem.textContent = tabpanelCloseText;
+        }
+    }
+
+    /**
+     * Listens for click events on items inside the main content area.
+     * It then calls the appropriate function based on the event target.
+     * This avoids adding multiple event handlers on individual elements.
+     */
+    mainContent.addEventListener('click', function(event) {
```

Comment:
Event delegation definitely has benefits for some scenarios, but you don't have that many events here and it does make the code a bit harder to read with everythign in a single handler like this. Not suggesting you change anythin now though, as this does all seem to work :)